tools/base/fpgad/config_file.c

```diff
@@ -30,7 +30,257 @@
 
 #include "config_file.h"
 
+#include <json-c/json.h>
 
+#ifdef LOG
+#undef LOG
+#endif
+#define LOG(format, ...) \
+log_printf("cfg: " format, ##__VA_ARGS__)
 
+int cfg_find_config_file(struct fpgad_config *c)
+{
+	char path[PATH_MAX];
+	char *e;
+	char *canon = NULL;
+	struct stat stat_buf;
 
+	e = getenv("HOME");
+	if (e) {
+		// try $HOME/.opae/fpgad.cfg
+		snprintf_s_s(path, sizeof(path),
+			     "%s/.opae/fpgad.cfg", e);
 
+		canon = canonicalize_file_name(path);
+		if (canon) {
+			if (!stat(canon, &stat_buf)) {
```

Comment:
I think canonicalize_file_name returns NULL if file does not exist

Reply:
fixed